### PR TITLE
xplat: shared library no longer requires an explicit call to dllmain

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

**Comment by obastemur** [comment]
Fixes #1811

File: bin/ChakraCore/ChakraCoreShared.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    JsRuntimeAttributes attr;
+    JsCreateRuntime(attr, nullptr, runtime);
+}
+#endif
+
+__attribute__ ((visibility ("default")))
+EXTERN_C BOOL WINAPI DllMain(HINSTANCE hmod, DWORD dwReason, PVOID pvReserved)
+{
+    // xplat doesn't need DllMain is being called.
+    // Instead, we handle attach / detach stuff internally
```
Comment (jianchun) [inline bin/ChakraCore/ChakraCoreShared.cpp]: Could you please add more comment here to describe where/how we handle those attach/detach  internally?

Comment (obastemur) [inline bin/ChakraCore/ChakraCoreShared.cpp]: Okay. Added the location of the file that is responsible from the internal mechanism. There are additional comments there.

**Comment by obastemur** [comment]
@jianchun Thanks for the review

File: bin/ChakraCore/ChakraCoreDllFunc.cpp [hunk 2]
```diff
@@ -19,26 +19,8 @@
...
-}
-#endif
-
 extern HANDLE g_hInstance;
-#ifdef _WIN32
 static ATOM  lockedDll = 0;
-#endif
-
-#ifdef _MSC_VER
-#define EXPORT_FUNC
```
Comment (Cellule) [inline bin/ChakraCore/ChakraCoreDllFunc.cpp]: The removal of this break nojit builds. At the end of this file there are 2 functions under `#if !ENABLE_NATIVE_CODEGEN` that are still using `EXPORT_FUNC`. I am not sure what the exact fix is, could you take a look please

**Pull request merged**